WIP Update UI colors (#895)
patternfly/patternfly-elements: docs/content/theme/color-palette.md

## Text & links

> Text and links are the most fundamental pieces of any web property, so they have their own variables aside from other UI elements. They also respond to context, such as a bit of text in a dark gray card would need to change to white. Or a link in a dark gray card would need to become a lighter shade of blue.  

Reviewer: ```suggestion
Text and links, being fundamental pieces of any web property, have their own set of variables. 
These variables assume a light background by default, assigning typography colors that pass accessibility when placed on white or very light gray contexts.  In addition, we provide contextual variables for these typographical elements for when they exist inside a dark or saturated context.  Text in a black card for example would need to flip to white to pass accessibility standards while links on a blue or red shade would need to update to white in order to be visible.
```

I found the wording above a bit difficult to understand. What would you think of something like this? ^

> You may choose to set the value of a link color to something in your brand, but it is not required, and these colors should provide a good user experience. 

Reviewer: ```suggestion
You can set the value of these typography color variables to something in your brand palette but be sure to validate them against a [contrast checker](https://webaim.org/resources/contrastchecker/) to ensure they continue to pass accessibility standards when set against your surface or UI palettes.  Aim for AAA status if you can! ⭐ 
```

Reviewer: I thought this added a little more guidance about how the variables relate to the palette and how to validate your updates are okay.

## Text & links

> Text and links, being fundamental pieces of any web property, have their own set of variables. 

Reviewer: TODO: We can remove broadcast text in the future and just have broadcast variables for links

Reviewer: We can address this in a separate PR: https://github.com/patternfly/patternfly-elements/issues/1009

 - **Information**: Charts, graphs, and wayfinding elements
 - **Hierarchy**: Showing structured order through color and typography

We've exposed 2 color variants for this design system to represent your brand:

 - Base
 - Accent

> These colors are used throughout PatternFly Elements, and **accent is the color which should stand out the most**. For example, if your brand colors are orange and medium gray, you'll want to set orange as the accent color. You'll see it appear on primary level call-to-action buttons and other elements that need to have more weight in the visual heirarchy of the page. 

Reviewer: ```suggestion
These colors are used throughout PatternFly Elements. **Accent** is the color which should stand out the most. For example, if your brand colors are orange and gray, we recommend you set orange as the accent color.  In doing this, that orange will now appear on primary level call-to-action buttons and other elements that need to have more weight in the visual hierarchy of the page. 
```

Reviewer: A few word-smithing ideas that I thought might simplify and clarify this point.

UI colors are meant to provide basic colors for other page elements besides links and body text. 

> If you are overriding these colors, you can do so by setting the CSS variables to have new values in the stylesheet of your page or app. You'll want to override the color itself, but also the corresponding hover variant and text color (that would be used if there was text on top of this color, like a button):

Reviewer: ```suggestion
If you are overriding these colors, you can do so by setting the CSS variables to have new values in the stylesheet of your page or app; we call this a `theme` layer. [Learn more about themes.](/theme) Override the color itself but also note its corresponding states such as hover and focus and its related text color (specifically for text that apperas on top of this color, such as in a button):
```

 - **Information**: Charts, graphs, and wayfinding elements
 - **Hierarchy**: Showing structured order through color and typography

> We've exposed 2 color variants for this design system to represent your brand:

Reviewer: Should we note that this is specific to UI since complement still exists for surfaces?

Author: updated, thanks

> We've exposed 2 color variants for this design system to represent your brand:

Author: ```suggestion
We've exposed 2 color variants for the UI elements in design system to represent your brand:
```

Author: what about this ^
